stream/src/main/java/org/apache/apex/malhar/stream/api/WindowedStream.java

```diff
@@ -51,6 +67,140 @@
  *
  * @since 3.4.0
  */
+@InterfaceStability.Evolving
 public interface WindowedStream<T> extends ApexStream<T>
 {
+
+  /**
+   * Count of all tuples
+   * @return new stream of Integer
+   */
+  <STREAM extends WindowedStream<Tuple.WindowedTuple<Long>>> STREAM count(Option... opts);
+
+  /**
+   * Count tuples by the key<br>
+   * @param name name of the operator
+   * @param convertToKeyValue The function convert plain tuple to k,v pair
+   * @return new stream of Key Value Pair
+   */
+  <K, STREAM extends WindowedStream<Tuple.WindowedTuple<KeyValPair<K, Long>>>> STREAM countByKey(Function.ToKeyValue<T, K, Long> convertToKeyValue, Option... opts);
+
+  /**
+   * Return top N tuples by the selected key
+   * @param N how many tuples you want to keep
+   * @param name name of the operator
+   * @param convertToKeyVal The function convert plain tuple to k,v pair
+   * @return new stream of Key and top N tuple of the key
+   */
+  <K, V, STREAM extends WindowedStream<Tuple.WindowedTuple<KeyValPair<K, List<V>>>>> STREAM topByKey(int N, Function.ToKeyValue<T, K, V> convertToKeyVal, Option... opts);
```

Comment:
Internally, that is added as a separate mapping operator, the reason we force it happen here is just because the KeyedWindowedOperator can only take input of type Tuple<KeyValuePair<>>, but compile cannot infer that from ApexStream/WindowedStream of type T. I have a plan to refactor every tuple in high-level api to key value pair tuple which has more flexibility.